### PR TITLE
build: migration and configs

### PR DESCRIPTION
**Pull request by christorange**
1. Migrated to Vite, there're troubles to use css module with TS on CRA and the configs of webpack config override are not stable. Besides, the starting and refreshing of app is much faster now.
2. Configured PostCSS to enable nested css.
3. Enabled auto import of css module styles, now we can write css just like normal but css module will be automatically enabled and can be used together with bootstrap.
4. Style updates, still on progress and will be sent by following commits.

### CODE REVIEW

**Comment by christorange** [comment]
Will setup Vitest for testing. Give me a while

**Review by IanSaucy — CHANGES_REQUESTED** [review]
I am getting the following error when running `npm run build`

```
HIC-ML-0033:se-d7-dashboard saucy$ npm run build

> d7-dashboard@0.1.0 build
> tsc && vite build

tsconfig.json:17:25 - error TS6046: Argument for '--moduleResolution' option must be: 'node', 'classic', 'node16', 'nodenext'.

17     "moduleResolution": "Bundler",
                           ~~~~~~~~~

tsconfig.json:18:5 - error TS5070: Option '--resolveJsonModule' cannot be specified without 'node' module resolution strategy.

18     "resolveJsonModule": true,
       ~~~~~~~~~~~~~~~~~~~


Found 2 errors in the same file, starting at: tsconfig.json:17
```

**Comment by christorange** [comment]
There's one unit test failed on the cloud but passed on my machine in multiple tries:

cloud:

image

local:

image

So I just uncommented that unit test, then I got 2 new firebase errors. Need help here, or maybe we just skip tests in CICD for now?

**Comment by IanSaucy** [comment]
I think the Firebase stuff will be fixed by Jason's incoming changes. Let's wait for those to merge and see how it goes.

**Comment by IanSaucy** [comment]
@christorange Can you merge in the new changes from `dev`?

**Comment by christorange** [comment]
four tests failed and all due to `getAuth()` method

image

**Comment by IanSaucy** [comment]
@christorange I think the only solution is to Mock the firebase functions we need to get the unit tests to work. So I'd probably comment the unit tests that are failing out for now. Also add a task on Trello that is about updating the unit tests to have mocking so that we can re-enable them.

**Comment by christorange** [comment]
@IanSaucy Finally passed CI/CD 😭

**Comment by christorange** [comment]
@IanSaucy Finally passed CI/CD 😭

File: .eslintrc.cjs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    project: true,
+    tsconfigRootDir: __dirname,
+  },
+  plugins: ['react-refresh'],
+  rules: {
+    'react-refresh/only-export-components': [
+      'warn',
+      { allowConstantExport: true },
+    ],
+    '@typescript-eslint/no-non-null-assertion': 'off',
```
Comment (IanSaucy) [inline .eslintrc.cjs]: can you leave this one on actually?

File: package.json [hunk 2]
```diff
@@ -3,27 +3,67 @@
...
+      "not op_mini all"
+    ],
+    "development": [
+      "last 1 chrome version",
+      "last 1 firefox version",
+      "last 1 safari version"
+    ]
+  },
+  "devDependencies": {
     "@patternfly/react-core": "^4.267.6",
```
Comment (IanSaucy) [inline package.json]: Are you sure this shouldn't be a regular dependency?

File: src/screens/login/Login.tsx [hunk 3]
```diff
@@ -55,34 +55,34 @@ const Login: React.FunctionComponent<Login> = (props) => {
...
+  //           navigate("/profile");
+  //         }
+  //       } else {
+  //         navigate("/profile");
+  //       }
+  //     })
+  //     .catch((e) => {
+  //       console.error(e);
+  //     });
+  // };
```
Comment (IanSaucy) [inline src/screens/login/Login.tsx:85]: Why did you remove this?

File: src/screens/welcome/Welcome.module.css [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  padding-top: 11vh;
+  padding-bottom: 8vh;
+  background: fit-content;
+}
+
+.welcome_title {
+  margin-left: 10vw;
+  font-style: normal;
+  font-weight: 600;
+  font-size: 36px;
```
Comment (IanSaucy) [inline src/screens/welcome/Welcome.module.css]: Probably should use `rem`

File: src/screens/welcome/Welcome.module.css [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  line-height: 40px;
```
Comment (IanSaucy) [inline src/screens/welcome/Welcome.module.css]: `rem` again?

File: src/screens/welcome/Welcome.module.css [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  display: flex;
+  flex-direction: column;
+  
+  .welcome_hint{
+    background-color: rgba(227, 184, 31, 0.5);
+    font-size: 20px;
+    margin-right: 20vw;
+    text-align: left;
+    padding-left: 25vw;
+    padding-top: 10px;
```
Comment (IanSaucy) [inline src/screens/welcome/Welcome.module.css:33]: Use `rem` here too?

**Review by IanSaucy — CHANGES_REQUESTED** [review]
Just some style changes

File: package.json (hunk 2 above)
Comment (christorange) [reply package.json]: deployed app of this pr works well but I'll move it to regular dependency

File: src/screens/login/Login.tsx (hunk 3 above)
Comment (christorange) [reply src/screens/login/Login.tsx:85]: The design removed 'login with google' button

File: src/screens/welcome/Welcome.module.css (hunk 6 above)
Comment (christorange) [reply src/screens/welcome/Welcome.module.css:33]: This is the inner padding of the bottom yellow box surrounding texts, I think it should be kept fixed but if you mean the font size of those texts I've changed it.

**Review by IanSaucy — APPROVED** [review]
lgtm
